lib/platform/azure/index.ts

```diff
@@ -620,11 +617,42 @@ export async function mergePr(
   );
 
   try {
-    await azureApiGit.updatePullRequest(
+    const response = await azureApiGit.updatePullRequest(
       objToUpdate,
       config.repoId,
       pullRequestId
     );
+
+    let retries = 0;
+    let isClosed = response.status === PullRequestStatus.Completed;
+    while (!isClosed && retries < 5) {
+      retries += 1;
+      const sleepMs = retries * 1000;
+      logger.trace(
+        `Updated PR ${pullRequestId} to closed status but change has not taken effect yet. Sleeping for ${sleepMs} before fetching PR again to check status (retry "${retries}")`
+      );
```

Comment:
```suggestion
      logger.trace(
        { pullRequestId, status: pr.status, retries },
        `Updated PR to closed status but change has not taken effect yet. Retrying ...`
      );
```

Use logger meta, maybe need prettify